tests/integration/test_push.py

```diff
@@ -0,0 +1,165 @@
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Tests for the push function."""
+
+# These tests are similar to workflow dispatch tests, in tests some duplication is ok to make the
+# tests easier to understand
+# pylint: disable=duplicate-code
+
+from uuid import uuid4
+
+import pytest
+from github.Branch import Branch
+
+from repo_policy_compliance import PushInput, Result, policy, push
+
+from .types_ import BranchWithProtection, RequestedCollaborator
+
+
+def test_invalid_policy():
+    """
+    arrange: given invalid policy
+    act: when push is called with the policy
+    assert: then a fail report is returned.
+    """
+    policy_document = {"invalid": "value"}
+
+    report = push(
+        input_=PushInput(
+            repository_name="repository 1",
+            branch_name="branch 1",
+            commit_sha="commit sha 1",
+        ),
+        policy_document=policy_document,
+    )
+
+    assert report.result == Result.FAIL, report.reason
+
+
+@pytest.mark.parametrize(
+    "github_branch, policy_enabled, expected_result",
+    [
+        pytest.param(
+            f"test-branch/push/branch-fail-enabled/{uuid4()}",
+            True,
+            Result.FAIL,
+            id="policy enabled",
+        ),
+        pytest.param(
+            f"test-branch/push/branch-fail-disabled/{uuid4()}",
+            False,
+            Result.PASS,
+            id="policy disabled",
+        ),
+    ],
+    indirect=["github_branch"],
+)
+def test_fail_branch(
```

Comment:
Shouldn't it be `test_branch_policy` as the expected result is not necessarily `fail` ?